pkg/machines/components/vm/firmwareModal.jsx

```diff
@@ -59,7 +59,7 @@ export class FirmwareModal extends React.Component {
                 <Modal.Body>
                     <Select.Select
                                    onChange={value => this.setState({ firmware: value })}
-                                   initial={this.props.firmware == 'efi' ? 'UEFI' : 'BIOS' }
+                                   initial={this.props.firmware == 'efi' ? this.props.firmware : 'bios' }
```

Comment:
Is that what the element actually expects? The commit message doesn't explain the "why", nor does this change anything else in the code that works with the `initial` property.

Reply:
Indeed I did not explain it at all - below follows explanation - let me know if I should repost for fixing the commit message.

So basically looking at the implementation of the pkg/lib/cockpit-components-select.jsx  that `initial` value is used to set the attribute of the `<select>` element here. https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/cockpit-components-select.jsx#L45

The value element should be one of the 'values' that are used for the options though. See https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/cockpit-components-select.jsx#L111

Previously I was using 'initial = "UEFI"' which was not any of the values of the SelectEntries (they use 'bios'/'efi') so the user would always see the pre-selected option in the dropdown to be the first - so BIOS.

This fixes the dropdown in matters of that if the already selected item is not the first from the list - when the dropdown is closed the shown item would be the already selected item.

I hope this makes sense.

Reply:
Thanks! That's what I suspected, but I'd rather have it confirmed.